java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -2739,6 +2740,114 @@ public void xadd_and_xtrim(BaseClient client) {
+                client
+                        .xadd(key2, Map.of(field2, field2 + "2"), StreamAddOptions.builder().id("2-2").build())
+                        .get();
+        assertNotNull(timestamp_2_2);
+
+        // setup third entries in streams key1 and key2
+        Map<String, String> timestamp_1_3_map = new LinkedHashMap<>();
+        timestamp_1_3_map.put(field1, field1 + "3");
+        timestamp_1_3_map.put(field3, field3 + "3");
+        String timestamp_1_3 =
+                client.xadd(key1, timestamp_1_3_map, StreamAddOptions.builder().id("1-3").build()).get();
+        assertNotNull(timestamp_1_3);
+
+        String timestamp_2_3 =
+                client
+                        .xadd(key2, Map.of(field2, field2 + "3"), StreamAddOptions.builder().id("2-3").build())
+                        .get();
+        assertNotNull(timestamp_2_3);
+
+        Map<String, Map<String, Map<String, String>>> result =
+                client.xread(Map.of(key1, timestamp_1_1, key2, timestamp_2_1)).get();
+
+        // check key1
+        assertEquals(
+                Map.of(
+                        timestamp_1_2,
+                        Map.of(field1, field1 + "2"),
+                        timestamp_1_3,
+                        Map.of(field1, field1 + "3", field3, field3 + "3")),
+                result.get(key1));
+
+        // check key2
+        assertEquals(
+                Map.of(
+                        timestamp_2_2,
+                        Map.of(field2, field2 + "2"),
+                        timestamp_2_3,
+                        Map.of(field2, field2 + "3")),
+                result.get(key2));
+
+        // Block on the final stream for 10 milliseconds
+        // and expect a null response because there is no incoming data
+        Map<String, Map<String, Map<String, String>>> blockedResult =
+                client
+                        .xread(
+                                Map.of(key1, timestamp_1_3, key2, timestamp_2_3),
+                                StreamReadOptions.builder().block(10L).build())
+                        .get();
+        assertNull(blockedResult);
+
+        // Key exists, but it is not a stream
+        assertEquals(OK, client.set(nonStreamKey, "bar").get());
+        ExecutionException executionException =
+                assertThrows(
+                        ExecutionException.class,
+                        () -> client.xread(Map.of(nonStreamKey, timestamp_1_3, key1, timestamp_2_3)).get());
+        assertInstanceOf(RequestException.class, executionException.getCause());
+
+        executionException =
+                assertThrows(
+                        ExecutionException.class,
+                        () -> client.xread(Map.of(key1, timestamp_1_3, nonStreamKey, timestamp_2_3)).get());
+        assertInstanceOf(RequestException.class, executionException.getCause());
+
+        // same-slot requirement
+        if (client instanceof RedisClusterClient) {
```

Comment:
We have another test case for that. If you need to rebase, update it after merging.